docs/design/openshift.md

```diff
@@ -191,4 +191,57 @@ $ cat release-manifests/image-references | head -n 10
 }
 ```
 
+## Update Image's Architecture ##
+
+The architecture of a release image is determined by the following algorithm:
+
+- check if it is a manifest list or an image by [media-types](https://github.com/openshift/docker-distribution/blob/main/docs/spec/manifest-v2-2.md#media-types). 
+
+  Here `skopeo` is chosen to show the information needed for the algorithm. It is up to the Cincinnati implementation for OpenShift how the information is retrieved. We may use `oc` (`oc image info` and `oc adm release info` to be precise) to achieve the same goal, but it would requires the understanding how `oc` gets the information. With `skopeo` (as some oc commands communicate with an OpenShift cluster), it is clear that all the information comes from the image.
+
+  ```console
+  ### manifest list
+  $ skopeo inspect --raw docker://quay.io/openshift-release-dev/ocp-release:4.15.5-multi | jq -r '.mediaType'
+  application/vnd.docker.distribution.manifest.list.v2+json
+
+  ### image manifest format
+  $ skopeo inspect --raw docker://quay.io/openshift-release-dev/ocp-release:4.15.5-s390x | jq -r '.mediaType'
+  application/vnd.docker.distribution.manifest.v2+json
+
+  $ skopeo inspect --config docker://quay.io/openshift-release-dev/ocp-release:4.18.5-s390x | jq -r '.architecture'
+  s390x
+  ```
+
+* If it is a manifest list, then the architecture is `multi`.
+
+* If it is an image, then the architecture is the one showed in the config like `s390x` in the last command above unless `release-manifests/release-metadata` [extracted from the release image](#update-image) indicates otherwise:
+
+  ```console
+  $ cat release-manifests/release-metadata | jq -r '.metadata."release.openshift.io/architecture"'
+  multi
+  ```
+
+  It is to rule out the noise from the multi's per-arch shards. E.g.,
+
+  ```console
+  ### get the digest for the ppc64le shard
+  $ skopeo inspect --raw docker://quay.io/openshift-release-dev/ocp-release:4.15.5-multi | jq -r '.manifests[]|select(.platform.architecture=="ppc64le")|.digest'
+  sha256:4a7c0f888b0a89157d7e9f48df55a361a9bfc009f63be560de79c24aa8acf8d0
```

Comment:
Stripping away even `skopeo` and going down to one cURL request:

```console
$ curl -sH 'Accept: application/vnd.docker.distribution.manifest.list.v2+json' https://quay.io/v2/openshift-release-dev/ocp-release/manifests/4.15.5-multi | jq -r '.manifests[] | select(.platform.architecture=="ppc64le").digest'
sha256:4a7c0f888b0a89157d7e9f48df55a361a9bfc009f63be560de79c24aa8acf8d0
```

Reply:
Done.

I cannot find the equivalent `curl` to `skopeo inspect --config`.
Otherwise, we can remove `skopeo` completely.

Reply:
The config is a blob referenced from the manifest:

```console
$ curl -sH 'Accept: application/vnd.docker.distribution.manifest.v2+json' https://quay.io/v2/openshift-release-dev/ocp-release/manifests/sha256:4a7c0f888b0a89157d7e9f48df55a361a9bfc009f63be560de79c24aa8acf8d0 | jq .config
{
  "mediaType": "application/vnd.docker.container.image.v1+json",
  "digest": "sha256:ba2c81793c38caa70f19a1e4ac1a960043dfd326936875e51c1fbab5d54e0b89",
  "size": 1829
}
$ curl -sL https://quay.io/v2/openshift-release-dev/ocp-release/blobs/sha256:ba2c81793c38caa70f19a1e4ac1a960043dfd326936875e51c1fbab5d54e0b89 | jq -r .architecture
ppc64le
```